overlord/ifacestate/ifacestate.go

```diff
@@ -101,22 +108,23 @@ func checkConnectConflicts(st *state.State, change *state.Change, plugSnap, slot
 	return nil
 }
 
-// ConnectOnInstall returns a set of tasks for (re)connecting an interface as part of snap installation (e.g. to reconnect interfaces on refresh, or autoconnect)
+// ConnectOnInstall returns a set of tasks for (re)connecting an interface as part of snap installation (e.g. to reconnect interfaces on refresh, or autoconnect).
+// This method doesn't check for conflicts, they need to be checked for upfront.
```

Comment:
given it does not conflicts check this shouldn't be exported I think,  we should just use connect directly